src/ArraySetops.chpl

```diff
@@ -0,0 +1,180 @@
+module ArraySetops
+{
+    use ServerConfig;
+
+    use BlockDist;
+    use SymArrayDmap;
+
+    use RadixSortLSD;
+    use Unique;
+    use Indexing;
+    use In1d;
+
+    use CommAggregation;
+
+    proc intersect1d(a1: [?aD] int, b1: [aD] int, assume_unique: string) {
+      if assume_unique == "False" {
+        var (a, _)  = uniqueSort(a1);
+        var (b, _)  = uniqueSort(b1);
+        var aux2 = concatset(a,b);
+        var aux_sort_indices = radixSortLSD_ranks(aux2);
```

Comment:
you could call the `radixSortLSD_keys` procedure to avoid the permutation on the next line. `radixSortLSD_keys` returns the sorted array if you don't need the permutation.